### ISSUE TITLE
Invalid characters in the file path installing app

### ISSUE THREAD

**Issue by amunoz-ldr** [microsoft/AL-Go#629]
Hello,

We have an issue with an extension due to a third-party dependency. We have configured al-go to install the dependent app, but when doing so, the following error occurs at the "Installing Apps" step:

`Error: RunPipeline action failed. Error: Caracteres no válidos en la ruta de acceso. Stacktrace: en <ScriptBlock>, C:\Users\AdminGH01\AppData\Local\Temp\23f9fcb5-08b0-4e37-93cb-0f2c0Error: RunPipeline action failed. Error: Caracteres no válidos en la ruta de acceso. Stacktrace: en <ScriptBlock>, C:\Users\AdminGH01\AppData\Local\Temp\23f9fcb5-08b0-4e37-93cb-0f2c0046e172\BcContainerHelper\HelperFunctions.ps1: línea 570 en CopyAppFilesToFolder, C:\Users\AdminGH01\AppData\Local\Temp\23f9fcb5-08b0-4e37-93cb-0f2c0046e172\BcContainerHelper\HelperFunctions.ps1: línea 525 en <ScriptBlock>, C:\Users\AdminGH01\AppData\Local\Temp\23f9fcb5-08b0-4e37-93cb-0f2c0046e172\BcContainerHelper\AppHandling\Run-AlPipeline.ps1: línea 1127 en <ScriptBlock>, C:\Users\AdminGH01\AppData\Local\Temp\23f9fcb5-08b0-4e37-93cb-0f2c0046e172\BcContainerHelper\AppHandling\Run-AlPipeline.ps1: línea 1101 en <ScriptBlock>, C:\Users\AdminGH01\AppData\Local\Temp\23f9fcb5-08b0-4e37-93cb-0f2c0046e172\BcContainerHelper\AppHandling\Run-AlPipeline.ps1: línea 1093 en <ScriptBlock>, C:\Users\AdminGH01\AppData\Local\Temp\23f9fcb5-08b0-4e37-93cb-0f2c0046e172\BcContainerHelper\AppHandling\Run-AlPipeline.ps1: línea 894 en Run-AlPipeline, C:\Users\AdminGH01\AppData\Local\Temp\23f9fcb5-08b0-4e37-93cb-0f2c0046e172\BcContainerHelper\AppHandling\Run-AlPipeline.ps1: línea 856 en <ScriptBlock>, C:\actions-runner\liderit-01\_work\_actions\microsoft\AL-Go-Actions\v3.1\RunPipeline\RunPipeline.ps1: línea 393 en <ScriptBlock>, C:\actions-runner\liderit-01\_work\_temp\9b4ccc5a-8d35-4247-8317-8acfefb28c2e.ps1: línea 2 en <ScriptBlock>, <sin archivo>: línea 1046e172\BcContainerHelper\HelperFunctions.ps1: línea 570 en CopyAppFilesToFolder, C:\Users\AdminGH01\AppData\Local\Temp\23f9fcb5-08b0-4e37-93cb-0f2c0046e172\BcContainerHelper\HelperFunctions.ps1: línea 525 en <ScriptBlock>, C:\Users\AdminGH01\AppData\Local\Temp\23f9fcb5-08b0-4e37-93cb-0f2c0046e172\BcContainerHelper\AppHandling\Run-AlPipeline.ps1: línea 1127 en <ScriptBlock>, C:\Users\AdminGH01\AppData\Local\Temp\23f9fcb5-08b0-4e37-93cb-0f2c0046e172\BcContainerHelper\AppHandling\Run-AlPipeline.ps1: línea 1101 en <ScriptBlock>, C:\Users\AdminGH01\AppData\Local\Temp\23f9fcb5-08b0-4e37-93cb-0f2c0046e172\BcContainerHelper\AppHandling\Run-AlPipeline.ps1: línea 1093 en <ScriptBlock>, C:\Users\AdminGH01\AppData\Local\Temp\23f9fcb5-08b0-4e37-93cb-0f2c0046e172\BcContainerHelper\AppHandling\Run-AlPipeline.ps1: línea 894 en Run-AlPipeline, C:\Users\AdminGH01\AppData\Local\Temp\23f9fcb5-08b0-4e37-93cb-0f2c0046e172\BcContainerHelper\AppHandling\Run-AlPipeline.ps1: línea 856 en <ScriptBlock>, C:\actions-runner\liderit-01\_work\_actions\microsoft\AL-Go-Actions\v3.1\RunPipeline\RunPipeline.ps1: línea 393 en <ScriptBlock>, C:\actions-runner\liderit-01\_work\_temp\9b4ccc5a-8d35-4247-8317-8acfefb28c2e.ps1: línea 2 en <ScriptBlock>, <sin archivo>: línea 1`

**Caracteres no válidos en la ruta de acceso** = "Invalid characters in the file path."

The app installs correctly on the client and in a development container as well. What could be different about the container created during the Al-Go process that causes the error? How can we identify which file contains the conflicting character so that we can notify the provider?

Thanks
Alejandro.

**Comment by freddydk** [comment]
Is this settings done in the installApps setting?

**Comment by amunoz-ldr** [comment]
Yes, the setting is:
  ```
"installApps" : [
    "dependencias\\*.zip"
]
```
The app is in a zip file.
With another apps in a zip file it works fine, it is just with this particular one.

**Comment by freddydk** [comment]
Does the name of the .zip file contain special characters, or the name of the app inside the .zip file?

**Comment by amunoz-ldr** [comment]
No, firts thing we did was to change the name of the app, now it's simply `gp.app`. It seems that has to do with the method `CopyAppFilesToFolder `and one or more particular files inside the app

**Comment by freddydk** [comment]
and what is the name of the .zip file?

**Comment by freddydk** [comment]
or did you rename the .zip file to gp.zip?
Any chance you could place gp.zip somewhere for me to download - then I can try to repro here?
you can email me a link at freddyk at microsoft dot com.

**Comment by amunoz-ldr** [comment]
It is gp.zip. I've already sent you a link.

**Comment by freddydk** [comment]
I was so convinced that this has to do with special characters, that I created a repo, using special characters and could repro a problem. This problem is fixed in the PR above from @mazhelez 

But..., this error was actually NOT caused by special characters (my bad) and the problem is actually not in AL-Go - it is in BcContainerHelper. Currently, it doesn't support wildcards in installApps lines (like dependencias/*.zip) 
It works if you use

"installApps" : [
    "dependencias\\Capture.zip",
    "dependencias\\gp.zip"
]

But... - I was also convinced that wildcards was supported, so I added support for this in the next version of BcContainerHelper preview, which will be available in 1-2 hours.
